force-app/recipes/classes/mocking/ReturnsThenThrows.cls

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+@isTest
+private class ReturnsThenThrows {
+  @isTest
+  static void recipe() {
+    // Arrange
+    final Mock deliveryServiceMock = Mock.forType(DeliveryService.class);
+    final MethodSpy planDeliverySpy = deliveryServiceMock.spyOn('planDelivery');
+    final Bakery sut = new Bakery((DeliveryService) deliveryServiceMock.stub);
+    planDeliverySpy.returns(Date.today().addDays(3));
+    planDeliverySpy.throwsException(new RecipeException());
```

Comment:
question: is it a specific use case in users tests ?

Reply:
Yes, it is to show the order of configuration matters